tests/qa_automation/kernel_multipath_flakyserver.pm

```diff
@@ -0,0 +1,108 @@
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: run multipath (I/O) tests on a multipathed iSCSI device while
+#          the iSCSI server keeps tampering with the LUNs (robustness test)
+# Maintainer: Klaus G. Wagner <kgw@suse.com>
+
+use base 'qa_run';
+use strict;
+use warnings;
+use lockapi;
+use testapi;
+use mmapi;
+use utils;
+use iscsi;
+
+sub start_testrun {
+    my $self = shift;
+
+    my $count;
+    # the mandatory parallel supportserver job which provides the multipathed test device
+    # Its job ID will be mentioned in the step info boxes associated with the subsequent mutex_wait() calls.
+    # _NOTE_: mutex_wait(), mutex_lock(), mutex_unlock(), mutex_try_lock() **need** this as second arg!
+    my $jobid_server = get_parents();
+    $jobid_server = $jobid_server->[0];
+
+    # START shameless copy from tests/qa_automation/kernel_multipath.pm
+    #
+    zypper_call("in open-iscsi qa_test_multipath");
+    systemctl 'start iscsid';
+    systemctl 'start multipathd';
+
+    # Set default variables for iscsi iqn and target
+    my $iqn    = get_var("ISCSI_IQN",    "iqn.2016-02.de.openqa");
+    my $target = get_var("ISCSI_TARGET", "10.0.2.1");
+    # Connect to iscsi server and obtain wwid for multipath configuration
+    iscsi_discovery $target;
+    iscsi_login $iqn, $target;
+    my $times = 10;
+    ($times-- && sleep 1) while (script_run('multipathd -k"show multipaths status" | grep active') == 1 && $times);
+    die 'multipath not ready even after waiting 10s' unless $times;
+    my $wwid = script_output("multipathd -k\"show multipaths status\" | grep active | awk {'print \$1\'}");
+    iscsi_logout $iqn, $target;
+
+    # Configure test suite with proper iscsi iqn and target
+    assert_script_run("sed -i '/^TARGET_DISK=.*/c\\TARGET_DISK=\"$iqn\"' /usr/share/qa/qa_test_multipath/data/vars");
+    assert_script_run("sed -i '/^TARGET=.*/c\\TARGET=\"$target\"' /usr/share/qa/qa_test_multipath/data/vars");
+    assert_script_run("cat /usr/share/qa/qa_test_multipath/data/vars");
+
+    ## # Configure wwid in configuration files for each test
+    ## my @config_files = qw(active_active active_passive path_checker_dio path_checker_tur);
+    ## foreach my $config_file (@config_files) {
+    # FIXME: current config files end up with three identical "multipath"
+    # entries for alias multipath-test. Clumsy and potentially detrimental.
+    # TODO: better check whether configfiles need editing in the first place.
+    # Beginning with qa_test_multipath-1.8-qa.2.1 this is supposed to
+    # no longer be needed.
+    #
+    ##     assert_script_run "sed -i '/wwid .*/c\\wwid $wwid' /usr/share/qa/qa_test_multipath/data/$config_file";
+    ##     assert_script_run("cat /usr/share/qa/qa_test_multipath/data/$config_file");
+    ## }
+    $self->qaset_config();
+    assert_script_run("/usr/share/qa/qaset/qaset reset");
+    #
+    # END  shameless copy from tests/qa_automation/kernel_multipath.pm
```

Comment:
@klwagner  you will have to move this shameless copy into function/method to qa_run to pass travis

Reply:
I see. Upon this I'm actually thinking of incorporating `kernel_multipath_flakyserver.pm` into `kernel_multipath.pm` altogether. Looks like this would only need a few case distinctions according to whether variable `ISCSI_MULTIPATH_FLAKY` is set. The code would possibly lose some elegance, though. I'll try this and, meanwhile, ask for maintainer pcervinka's opinion.

Reply:
Done in [301cf8b](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8539/commits/301cf8bcc708428c480a1bc474280b836f57ed97) below. Still WIP, of course, but at least no formal flaws are left by now. Also notified pcervinka in case he wants to review.